examples/test_util/EclFile.cpp

```diff
@@ -16,490 +16,835 @@
 
 namespace {
-  
-bool isEOF(std::fstream *fileH){
-    
-    /* try to read 4 byte (integer), if passed eof of, return true.
-       if not, seek back to current file possition  */
-    
-    int num;
-    long int pos=fileH->tellg();
-    fileH->read((char *)&num, sizeof(num));
-    
-    if (fileH->eof()){
-        return true;
-    } else {
-        fileH->seekg (pos);
-        return false;
-    }
-}
-
-int flipEndianInt(const int &num) {
-
-   unsigned int tmp=__builtin_bswap32(num);
-   
-   return (int)tmp;
-}
-
-float flipEndianFloat(const float &num) {
 
-    float value=num;
 
-    char *floatToConvert = reinterpret_cast<char*>(&value);
-    std::reverse(floatToConvert, floatToConvert+4);
-    
-    return value;
-}
+bool isFormatted(const std::string& filename)
+{
+    int p = filename.find_last_of(".");
+    int l = filename.length();
 
-double flipEndianDouble(const double &num) {
-    double value=num;
-    
-    char *doubleToConvert = reinterpret_cast<char*>(&value);
-    std::reverse(doubleToConvert, doubleToConvert+8);
+    std::string rootN = filename.substr(0,p);
+    std::string extension = filename.substr(p,l-p);
 
-    return value;
+    return extension.substr(1,1) == "F";
 }
 
-std::tuple<const int, const int> block_size_data(EIOD::eclArrType arrType){
-
-    switch(arrType) {
-        case EIOD::INTE  : return std::make_tuple(EIOD::sizeOfInte,EIOD::MaxBlockSizeInte);   break;
-        case EIOD::REAL  : return std::make_tuple(EIOD::sizeOfReal,EIOD::MaxBlockSizeReal);   break;
-        case EIOD::DOUB  : return std::make_tuple(EIOD::sizeOfDoub,EIOD::MaxBlockSizeDoub);   break;
-        case EIOD::LOGI  : return std::make_tuple(EIOD::sizeOfLogi,EIOD::MaxBlockSizeLogi);   break;
-        case EIOD::CHAR  : return std::make_tuple(EIOD::sizeOfChar,EIOD::MaxBlockSizeChar);   break;
-        case EIOD::MESS  : throw std::invalid_argument("Type 'MESS' have not assosiated data") ;   break;
-        default          : throw std::invalid_argument("Unknown type"); break;
-    }
-}
 
-std::string trimr(const std::string &str1) {
+bool isEOF(std::fstream* fileH)
+{
+    int num;
+    long int pos = fileH->tellg();
+    fileH->read(reinterpret_cast<char*>(&num), sizeof(num));
 
-    std::string tmpStr=str1;
-    
-    while (tmpStr.back()==' '){
-        tmpStr.pop_back();
+    if (fileH->eof()) {
+        return true;
+    } else {
+        fileH->seekg (pos);
+        return false;
     }
-
-    return std::move(tmpStr);
 }
 
 
-void readBinaryHeader(std::fstream &fileH, std::string &arrName, int &size, EIOD::eclArrType &arrType){
-
+void readBinaryHeader(std::fstream& fileH, std::string& arrName,
+                      int& size, EIOD::eclArrType &arrType)
+{
     int bhead;
     std::string tmpStrName(8,' ');
     std::string tmpStrType(4,' ');
 
-    if (!fileH.is_open()){
-        std::string message="fstream fileH not open for reading";
-        throw std::runtime_error(message);
+    if (!fileH.is_open()) {
```

Comment:
As long as the `EclFile` class manages the `fileH`it is maybe a bit over the top to repeatedly check if it is open?

Reply:
done.